### PR TITLE
Frontend implementation for adding comments to an application

### PR DESCRIPTION
**Pull request by llmlks**

### CODE REVIEW

File: client/src/redux/sagas/applicationSaga.js [hunk 1]
```diff
@@ -120,10 +120,43 @@ function* getApplicants() {
...
+
+    if (response.status === 201) {
+      const applicants = yield select(getCurrentApplicants)
+
+      const commentedApplicant = applicants
+        .reduce((applicants, stage) => applicants.concat(stage.applicants), [])
+        .find(applicant => applicant.id === payload.applicationId)
+
+      if (!commentedApplicant.comments) {
+        commentedApplicant.comments = []
```
Comment (josalmi) [inline client/src/redux/sagas/applicationSaga.js]: Mutable operation

File: client/src/redux/sagas/applicationSaga.js [hunk 2]
```diff
@@ -120,10 +120,43 @@ function* getApplicants() {
...
+      }
+
+      commentedApplicant.comments.push(response.data)
```
Comment (josalmi) [inline client/src/redux/sagas/applicationSaga.js]: Mutable operation

File: client/src/components/applicants/ApplicantModal.js [hunk 3]
```diff
@@ -1,20 +1,55 @@
 import React from 'react'
+import { connect } from 'react-redux'
 import PropTypes from 'prop-types'
-import { Button } from '@material-ui/core'
-import Modal from '@material-ui/core/Modal'
+import { Button, Modal, TextField } from '@material-ui/core'
```
Comment (happtur) [inline client/src/components/applicants/ApplicantModal.js]: https://material-ui.com/guides/minimizing-bundle-size/ (?)

File: client/src/redux/sagas/applicationSaga.js [hunk 4]
```diff
@@ -142,12 +142,23 @@ function* addComment({ payload }) {
...
 
+        let newStage = stage
         if (commentedApplicant) {
           commentedApplicant.comments = commentedApplicant.comments
             ? [ ...commentedApplicant.comments, response.data ]
-            : []
+            : [ response.data ]
+
+          const notCommentedApplicants = [
+            ...stage.applicnats.filter(applicant =>
```
Comment (happtur) [inline client/src/redux/sagas/applicationSaga.js]: applicnats :P

File: client/src/redux/sagas/applicationSaga.js [hunk 5]
```diff
@@ -120,10 +121,59 @@ function* getApplicants() {
...
+    }
+
+    const response = yield call(jobApplicationApi.addComment, data)
+
+    if (response.status === 201) {
+      const stages = yield select(getStages)
+
+      const newStages = stages.map(stage => {
+        const commentedApplicant = {
+          ...stage.applicants.find(applicant =>
```
Comment (happtur) [inline client/src/redux/sagas/applicationSaga.js]: Something odd happens here (?)  
even when the find method returns undefined (i.e. stage does not contain the application that got a comment), commentedApplicant is { comments: [{ theNewComment }] }. So this "applicant" is added to all other stages than the one that should be changed (that stage is updated correctly)

Comment (happtur) [reply client/src/redux/sagas/applicationSaga.js]: And the comments are in "applicationComments", not "comments", when fetched. Comments might be better, but...

File: client/src/redux/sagas/applicationSaga.js [hunk 6]
```diff
@@ -120,10 +121,59 @@ function* getApplicants() {
...
+            applicant.id === payload.applicationId
```
Comment (happtur) [inline client/src/redux/sagas/applicationSaga.js]: Something odd happens here (?)  
even when the find method returns undefined (i.e. stage does not contain the application that got a comment), commentedApplicant is { comments: [{ theNewComment }] }. So this "applicant" is added to all other stages than the one that should be changed (that stage is updated correctly)

Comment (happtur) [reply client/src/redux/sagas/applicationSaga.js]: And the comments are in "applicationComments", not "comments", when fetched. Comments might be better, but...

**Pull request merged**